cli/transfer.go

```diff
@@ -0,0 +1,81 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/coder/coder/cli/clibase"
+	"github.com/coder/coder/cli/cliui"
+	"github.com/coder/coder/codersdk"
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+)
+
+func (r *RootCmd) transferWorkspace() *clibase.Cmd {
+	var (
+		workspaceID string
+		toUserID    string
+	)
+	client := new(codersdk.Client)
+	cmd := &clibase.Cmd{
+		Use:   "transfer --workspace <workspaceId> --owner <userId>",
+		Short: "Will transfer ownership of a workspace to a different user.",
+		Middleware: clibase.Chain(
+			clibase.RequireNArgs(0),
+			r.InitClient(client),
+		),
+		Handler: func(inv *clibase.Invocation) error {
+			if workspaceID == "" {
+				w, err := cliui.Prompt(inv, cliui.PromptOptions{
+					Text: "Workspace Id:",
+				})
+				if err != nil {
+					return err
+				}
+				workspaceID = w
+			}
+			if toUserID == "" {
+				u, err := cliui.Prompt(inv, cliui.PromptOptions{
+					Text: "To User Id:",
```

Comment:
Can you a think of a way to make this easier to use if there are, let's say, 1000+ users? Ditto with line 29 above -- can you think of how to make this easier to use if you have a lot of workspaces?

Reply:
Sure! Right now it's using the pattern that other CLI operations seem to use. If a CLI flag is passed it uses it, and if it isn't then it prompts for those values. And while the argument flags could be a way to automate things if you knew the workspaces you wanted to transfer, I'm guessing you're looking at it more form a "transfer user X to user Y" type scenario. In that case I'd probably create an entirely new feature to transfer all of a user's resources to another, instead of just a single workspace. Admittedly I'm not familiar with the day to day needs of the Coder users to make a call if both of these features would be needed, or if the bulk "user transfer" would suffice.